components/site/src/sass/mod.rs

```diff
@@ -52,6 +55,24 @@ pub fn compile_sass(base_path: &Path, output_path: &Path) -> Result<()> {
     Ok(())
 }
 
+/// write out a subset of the Zola config document to a temporary SCSS file
+/// as an SCSS map variable literal.  this will allow parts of the site's
+/// config to be usable during Sass compilation.  this enables theme configuration
+/// like allowing the site owner to change header color.  this function returns
+/// a tempdir holding a single `.scss` file.  the tempdir should then be used as
+/// a load directory above when compiling the site's Sass files.  the tempdir
+/// and contained `.scss` file will be deleted on drop of the returned `TempDir`
+/// struct, which should happen after Sass compilation finishes.
+fn build_dependencies_dir_from_config(config: &Config) -> Result<TempDir> {
+    let dir = tempdir().context("failed to create tempdir for SASS dependencies")?;
+
+    let config_serialized = serde::serialize_config(config)?;
+
+    std::fs::write(dir.path().join("zola.scss"), format!("$config: {}", config_serialized))?;
+
+    Ok(dir)
```

Comment:
What do you think of writing it to the sass directory so people can easily see what there is? It would be still overwritten on each serve/build but we can maybe not delete it

Reply:
you're saying write it to the user's sass source directory?  i figured that would be undesirable because then maybe it gets erroneously checked in.  but i'm open to it if that seems better.

Reply:
I think it would be ok to check it in? It's not going to change very often

Reply:
It's fine to not add it the directory but it's a bit annoying that you can't see it to look at what exactly is defined

Reply:
any suggestions on how to surface this content to the site builder without putting it in the directory?  i could create a non-tmp directory in `/tmp` and log the location of it?

Reply:
Honestly I'm not too sure. Maybe it's ok to just show an example of how it looks like in the docs